packages/contracts/registry/README.md

```diff
@@ -0,0 +1,83 @@
+
+Only the owner can insert new currencies and amounts.
+Every time a contract is added to the registry it is added both to the map and the allowlist set. If the contract is removed from the map - be it because HYC frontend doesn't want to support it anymore or because it has been deprecated in an upgrade - it is still going to be available in the allowlist, to allow users to withdraw funds they had previously deposited to them.
+In case a contract ever needs to be removed from the allowlist, this operation should be explicitly done after removing from the map, using `remove_from_allowlist`.
+
+## Security assumptions
+This contract does not handle user funds directly, it is only an on-chain information registry. Therefore, this contract has no intention of being run as a fully decentralized application. It is going to be controlled by the development team during project's early phases and handled over to the DAO structure later on.
+
+If a hacker ever compromises this contract, the main consequence would be the possibility to insert malicious contract into the registry, which would be trusted by frontends and relayers running the protocol - and by its end users in extension. This could carry great consequences, allowing phishing scams to effectivelly attack users. However, there would be no risk for already deposited funds.
+
+To keep maximum security, owner's private keys should be stored in a hardware wallet and later on transferred to a DAO structure.
+
+## Contract interface
+
+### Initialize
+1. `new`
+params:
+currency: Currency, amount: U256, contract: AccountId
+  - owner: AccountId -> account that will be able to edit the registry
+
+Method to initialize the contract with a specified owner. The owner is going to be the only account able to alter the registry. Current implementation does not allow future changes in owner.
+
+### Change methods (owner restricted)
+1. `add_entry`
+params:
+  - currency: Currency -> The currency of the contract address you want to add
+  - amount: U256 -> Amount value of deposits for HYC contract you want to add
+  - account_id: AccountId -> Address of the HYC contract that you're adding to the registry
+
+This method can only be called by the contract owner.
+It adds a specific entry to the registry (adds a contract address for the (currency, amount) pair).
+If there was a previous contract stored in the location it will be overwritten.
+
+2. `remove_entry`
+params:
+  - currency: Currency -> The currency of the contract address you want to remove
+  - amount: U256 -> Amount value of deposits for HYC contract you want to remove
+
+This method removes one entry from the registry, according to the (currnecy, amount) pair specified in args.
+Panics if currency or amount is not registered.
+
+3. `remove_from_allowlist`
+params:
```

Comment:
Since it's possible to remove from the allowlist, don't you think we should hava a guide on how to withdraw manually if the contract was removed?